command/filemanager.go

```diff
@@ -0,0 +1,50 @@
+package command
+
+import (
+	"path"
+
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/pulumi/pulumi-command/sdk/go/command/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	linuxTempDir = "/tmp"
+)
+
+type FileManager struct {
+	runner *Runner
+}
+
+func NewFileManager(runner *Runner) *FileManager {
+	return &FileManager{
+		runner: runner,
+	}
+}
+
+func (fm *FileManager) CreateDirectory(name string, remotePath pulumi.StringInput, useSudo bool, opts ...pulumi.ResourceOption) (*remote.Command, error) {
+	return fm.runner.Command(name, pulumi.Sprintf("mkdir -p %s", remotePath), nil, pulumi.Sprintf("rm -rf %s", remotePath), nil, useSudo, opts...)
+}
+
+func (fm *FileManager) TempDirectory(name string, opts ...pulumi.ResourceOption) (*remote.Command, string, error) {
+	tempDir := path.Join(linuxTempDir, name)
+	folderCmd, err := fm.CreateDirectory("tmpdir-"+name, pulumi.String(tempDir), false, opts...)
+	return folderCmd, tempDir, err
+}
+
+func (fm *FileManager) CopyFile(localPath, remotePath string, opts ...pulumi.ResourceOption) (*remote.CopyFile, error) {
+	// fileHash, err := utils.FileHash(localPath)
+	// if err != nil {
+	// 	return nil, err
+	// }
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
Leftover ?
```suggestion
```